src/object/schedcontext.c

```diff
@@ -110,6 +112,18 @@ static exception_t decodeSchedContext_Bind(sched_context_t *sc, extra_caps_t ext
             return EXCEPTION_SYSCALL_ERROR;
         }
 
+        switch (thread_state_get_tsType(TCB_PTR(cap_thread_cap_get_capTCBPtr(cap))->tcbState)) {
+        case ThreadState_BlockedOnReceive:
+        case ThreadState_BlockedOnSend:
+        case ThreadState_BlockedOnNotification:
+        case ThreadState_BlockedOnReply:
```

Comment:
recommend a short handle for this (`isBlocked`)

Reply:
I'll need to come up with a different name, `isBlocked` already exists but is 'private' to thread.c and classifies slightly differently.